src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +62,25 @@
     }
   }
 
+  /**
+   * Get the last traded price for each Symbol that is passed in.
+   *
+   * @param symbol the symbol to get a historical prices for.
+   * @param range the range to get historical prices for
+   * @param date the date to get historical prices for
+   * @return a list of historical prices for the Symbol that is passed in.
+   */
+  public List<IexHistoricalPrices> getHistoricalPricesForSymbol(
+      final String symbol,
+      final String range,
+      final String date) {
+    if (symbol.compareTo("") == 0) {
+      throw new ResponseStatusException(
+          HttpStatus.NOT_FOUND, "entity not found"
+      );
+    }
```

Comment:
This implements the http error status we discussed. However, consider if this method is called from somewhere other than the Controller. If there is an invalid input to this method, it will throw an HTTP status exception, even though there was no HTTP call. You need to remove the concern that this will result in an HTTP error coming from the controller from this method. The controller knows about HTTP and any validation, this method should confine its responsibility to validating the inputs provided and throw an exception that's scoped correctly for its context.

Reply:
Also, shouldn't there be validation on the other fields as well?

Reply:
> Also, shouldn't there be validation on the other fields as well?

No I don't believe so. Range is defaulted to 1m by both our code and the cloud API so that won't be an issue. Date is not required by either API either so no error there if empty